src/ert/scheduler/lsf_driver.py

```diff
@@ -82,17 +85,37 @@ class RunningJob(BaseModel):
 JOB_ALREADY_FINISHED_BKILL_MSG = "Job has already finished"
 
 
-class _Stat(BaseModel):
-    jobs: Mapping[str, AnyJob]
+def _create_job_class(job_dict: Mapping[str, str]) -> AnyJob:
+    job_state = job_dict["job_state"]
+    if job_state in get_type_hints(FinishedJobSuccess)["job_state"].__args__:
```

Comment:
This smells a little bit. Can we get away with just a dictionary from string job-state to the correct class?

Reply:
Yes, it looks way better but Mypy does not like that whatsoever. 
```
src/ert/scheduler/lsf_driver.py:98: error: Argument 1 to "FinishedJobSuccess" has incompatible type "Literal['EXIT', 'DONE', 'PEND', 'RUN', 'ZOMBI', 'PDONE', 'SSUSP', 'USUSP', 'PSUSP', 'UNKWN']"; expected "Literal['DONE', 'PDONE']"  [arg-type]
src/ert/scheduler/lsf_driver.py:98: error: Argument 1 to "FinishedJobFailure" has incompatible type "Literal['EXIT', 'DONE', 'PEND', 'RUN', 'ZOMBI', 'PDONE', 'SSUSP', 'USUSP', 'PSUSP', 'UNKWN']"; expected "Literal['EXIT', 'ZOMBI']"  [arg-type]
src/ert/scheduler/lsf_driver.py:98: error: Argument 1 to "QueuedJob" has incompatible type "Literal['EXIT', 'DONE', 'PEND', 'RUN', 'ZOMBI', 'PDONE', 'SSUSP', 'USUSP', 'PSUSP', 'UNKWN']"; expected "Literal['PEND']"  [arg-type]
src/ert/scheduler/lsf_driver.py:98: error: Argument 1 to "RunningJob" has incompatible type "Literal['EXIT', 'DONE', 'PEND', 'RUN', 'ZOMBI', 'PDONE', 'SSUSP', 'USUSP', 'PSUSP', 'UNKWN']"; expected "Literal['RUN', 'SSUSP', 'USUSP', 'PSUSP']"  [arg-type]
src/ert/scheduler/lsf_driver.py:98: error: Argument 1 to "IgnoredJobstates" has incompatible type "Literal['EXIT', 'DONE', 'PEND', 'RUN', 'ZOMBI', 'PDONE', 'SSUSP', 'USUSP', 'PSUSP', 'UNKWN']"; expected "Literal['UNKWN']"  [arg-type]
```